src/app/exam/Examinees.js

```diff
@@ -6,62 +6,81 @@ import {examService} from "../../services/services";
         const [showAddGroupModal, setShowAddGroupModal] = useState(false);
         const [showRemoveExamineeConfirmationModal, setShowRemoveExamineeConfirmationModal] = useState(false);
 
         const examId = match.params.examId;
 
+        useEffect(() => {
+            if (!exam) {
+                examService.getExam(examId)
+                    .then(exam => setExam(exam))
+            }
+            if (!examinees && exam) {
+                console.log("in effect")
+                examService.getExaminees(examId)
+                    .then(students => setExaminees(students));
+            }
+        }, [exam, examId, examinees]);
+
         const actionItemsButton = ({examinee}) => new ThreeDotsButton({
-            dropDownItems: [
-                {
-                    name: "Remove",
-                    dangerous: true,
-                    onClick: () => {
-                        setShowRemoveExamineeConfirmationModal(true)
-                        setExaminee(examinee)
-                    }
+            dropDownItems: [{
+                name: "Remove",
+                dangerous: true,
+                onClick: () => {
+                    setShowRemoveExamineeConfirmationModal(true)
+                    setSelectedExaminee(examinee)
                 }
-            ]
+            }]
         })
 
-        const addExaminees = (emails) => {
-            examService.addExaminees(exam.id, emails)
-                .then(errorList => console.log(`errorList=${errorList}`))
+        const addExaminees = async (emails) => {
+            await examService.addExaminees(exam.id, emails)
+                .then(errorList => {
+                    console.log(`errorList=${errorList}`);
+                })
+            examService.getExaminees(examId)
+                .then(students => {
+                    // TODO: remove setting empty array and fixed "render more hooks than expected problems"
+                    setExaminees([])
+                    setExaminees(students)
+                });
         }
 
-        const removeExaminee = (emails) => {
-            examService.deleteExaminees(examId, emails).then()
+        const removeExaminee = (email) => {
+            examService.deleteExaminees(examId, email)
+                .then(() => removeExamineeByEmail(email));
         }
 
-        useEffect(() => {
-            if (!exam) {
-                examService.getExam(examId)
-                    .then(exam => setExam(exam))
-            }
-            if (!students && exam) {
-                examService.getExaminee(examId)
-                    .then(students => setStudents(students));
+        function removeExamineeByEmail(email) {
+            const removed = examinees.find(examinee => examinee.email === email);
+            const index = examinees.indexOf(removed);
+            if (index > -1) {
+                examinees.splice(index, 1);
             }
```

Comment:
add a method in `/utils/array.js`

```js
function remove(array, objToRemove) {
    const index = array.indexOf(objToRemove);
    if (index > -1) {
        array.splice(index, 1);
    }
}
```

and please, 
```
function removeIf(array, predicate) {
...
}
```

so you can write `removeIf(examinees, examinee => examinee.email === email)`